GoogleForCreators/web-stories-wp
# Redesign: TextArea component

## Summary
- Adds new TextArea component
- Refactors the code a bit to use some shareable parts from the `Input` component
- Uses the component for the Image accessibility panel to ensure it works.


## Relevant Technical Choices
Since the style and behaviour are quite similar to the `Input` then initially it seemed like it might be a sub-input under the `Input` but it's still a different element and uses different props (e.g. can't have a suffix, supports the character count) so in the end it didn't seem a good idea and `TextArea` seems to make sense as a separate component.

This PR, however, does move the shareable parts to `utils` folder.


## To-do

- [ ] Verify that tests sill work



## User-facing changes
| Before | After |
|--------|------|
| ![Screenshot 2021-03-09 at 16 07 23](https://user-images.githubusercontent.com/3294597/110482665-303d6f80-80e9-11eb-9977-416d55f67018.png) | ![Screenshot 2021-03-09 at 16 05 57](https://user-images.githubusercontent.com/3294597/110482698-37fd1400-80e9-11eb-91c0-54f8b0e76216.png) |


## Testing Instructions
In addition to the steps below for QA, please test this also in the storybook under TextArea component.


### QA






This PR can be tested by following these steps:

1. Add an image and see the Accessibility panel
2. Verify the alt textarea looks as in [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=742%3A23319)

### UAT







This PR can be tested by following these steps:

1.

## Checklist




---



Fixes #6495 
Fixes #6465
